src/coap_resource.c

```diff
@@ -985,21 +984,67 @@ coap_delete_observer(coap_resource_t *resource, coap_session_t *session,
                    (void *)s, outbuf, s->cache_key->key[0], s->cache_key->key[1],
                    s->cache_key->key[2], s-> cache_key->key[3]);
   }
-  if (s && session->context->observe_deleted)
+  if (session->context->observe_deleted)
     session->context->observe_deleted(session, s,
                                       session->context->observe_user_data);
 
-  if (resource->subscribers && s) {
+  if (resource->subscribers) {
     LL_DELETE(resource->subscribers, s);
     coap_session_release_lkd(session);
     coap_delete_pdu(s->pdu);
     coap_delete_cache_key(s->cache_key);
     coap_free_type(COAP_SUBSCRIPTION, s);
   }
 
+  return;
+}
+
+int
+coap_delete_observer(coap_resource_t *resource, coap_session_t *session,
+                     const coap_bin_const_t *token) {
+  coap_subscription_t *s;
+
+  s = coap_find_observer(resource, session, token);
+  if (s)
+    coap_delete_observer_internal(resource, session, s);
+
   return s != NULL;
 }
 
+int
+coap_delete_observer_request(coap_resource_t *resource, coap_session_t *session,
+                             const coap_bin_const_t *token, coap_pdu_t *request) {
+  coap_subscription_t *s;
+  int ret = 0;
+
+  s = coap_find_observer(resource, session, token);
+  s = NULL;
```

Comment:
Hi @mrdeep1, by chance I just noticed that `s` is set to `NULL` here right after `coap_find_observer` has been called – is that intentional or could that be a bug?

Reply:
Oops - I had added that in to test functionality and omitted to remove it. See #1444 for the fix.

Much appreciated you finding and reporting it.